lib/helpers/utils.poku.js

```diff
@@ -6768,14 +6770,142 @@ const testCases = [
+  ["http://user@example.com/path", true], // Valid with user only
+  // IRI with IPv6 literal (tests authority parsing)
+  ["http://[2001:db8::1]:8080/path", false], // Valid IPv6
+  ["http://[2001:db8::1]/path", false], // Valid IPv6 without port
+  // Potentially problematic characters in path/query/fragment (if not already covered)
+  ["http://example.com/path with spaces", false], // Space not encoded
+  ["http://example.com/path<with>brackets", false], // < > not typically allowed unencoded
+  ['http://example.com/path"with"quotes', false], // " not typically allowed unencoded in URI/IRI ref
+  // Test case sensitivity for scheme check (uses original `iri`)
+  ["HTTP://example.com", true], // Scheme case (URL constructor should handle)
+  ["HTTPS://EXAMPLE.COM/PATH", true],
+  // Edge case: IRI that is just a scheme
+  ["mailto:", false],
+  ["https:", false],
+  ["http:", false],
+  // Re-test specific percent-encoding edge case mentioned in comments
+  ["http://example.com/path%ab%cd%ef", true], // Valid percent encodings
+  ["http://example.com/path%ab%cd%e", false], // Invalid: incomplete %e at end
+  ["http://example.com/path%ab%cd%eg", false], // Invalid: %eg
+  ["http://example.com/path%ab%cd%", false], // Invalid: trailing %
+  ["http://example.com/path%ab%cd%0", false], // Invalid: %0
+  ["http://example.com/path%ab%cd%0Z", false], // Invalid: %0Z (Z is hex, but makes the sequence too long if interpreted as %ab%cd%0Z)
+  // Test with extremely long, but valid, percent-encoded sequence (pushes validateIri/URL)
+  // This string is valid UTF-8 percent-encoded 'A' repeated many times.
+  // encodeURIComponent("A".repeat(10000)) produces a very long string of %41
+  // Let's simulate a long valid percent-encoded part manually for a simpler test
+  ["http://example.com/data/" + "%41%42%43%44".repeat(10000), true], // Repeats 'ABCD' encoded
+  // UNC Paths (IRI references)
+  // Standard UNC path (often treated as URIs like \\server\share\path -> file://server/share/path or \\server\share -> smb://server/share)
+  // However, as IRI *references* starting with \\, they are generally invalid unless specifically scheme-less references
+  // The IRI spec defines scheme-less references as relative. \\server is not a valid relative path segment.
+  ["\\\\server\\share\\path\\file.txt", false], // Looks like UNC, invalid as IRI ref
+  ["file://server/share/path/file.txt", true], // Correct URI form if that's the intent
+  // UNC path with spaces (invalid as IRI ref, valid file URI)
+  ["\\\\server name\\share name\\file name.txt", false],
+  ["file://server%20name/share%20name/file%20name.txt", true],
+  // UNC path with Unicode (invalid as IRI ref, valid file URI if percent-encoded)
+  ["\\\\サーバー\\共有\\ファイル.txt", false], // Raw Unicode UNC - invalid IRI ref
+  // Correct IRI for UNC-like path would need a scheme, e.g., file:
+  [
+    "file:///%E3%82%B5%E3%83%BC%E3%83%90%E3%83%BC/%E5%85%B1%E6%9C%89/%E3%83%95%E3%82%A1%E3%82%A4%E3%83%AB.txt",
```

Comment:
I have no idea why this is valid. There could be some amount of invalid IRIs that might escape the new implementation. Please file those as bugs, so that we can improve the parser.